lib/box2d/sprite_body_component.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:ui';
+
+import '../anchor.dart';
+import '../components/component.dart';
+import '../sprite.dart';
+import 'body_component.dart';
+import 'box2d_game.dart';
+
+abstract class SpriteBodyComponent extends BodyComponent {
+  SpriteComponent spriteComponent;
+  final double width;
+  final double height;
+
+  /// Make sure that your [width] and [height] of the sprite
+  /// matches the bounding rectangle of the body created by createBody()
+  SpriteBodyComponent(
+    Sprite sprite,
+    this.width,
+    this.height,
+    Box2DGame game,
+  ) : super(game) {
+    spriteComponent = SpriteComponent.fromSprite(width, height, sprite)
+      ..anchor = Anchor.center;
+  }
+
+  @override
+  bool loaded() => body.isActive() && spriteComponent.loaded();
+
+  @override
+  void render(Canvas c) {
+    super.render(c);
+    if (spriteComponent.loaded()) {
+      final screenPosition = viewport.getWorldToScreen(body.position);
```

Comment:
If this one is done in update() the rendering of the sprite lags behind the rendering of the fixtures more and more the more speed the body has. Any clue why this is?

Reply:
I assume it's an order of operations problem, if make this on the update when does it happen in relation to the box2d update? should probably be box2d update -> your viewport conversion -> render